mkosi/__init__.py

```diff
@@ -2166,6 +2174,24 @@ def finalize_cmdline(context: Context, roothash: Optional[str]) -> list[str]:
     return cmdline + context.config.kernel_command_line
 
 
+def finalize_initrds(context: Context) -> list[Path]:
+    if any((context.artifacts / "io.mkosi.microcode").glob("*")):
+        initrds = sorted((context.artifacts / "io.mkosi.microcode").iterdir())
+    elif microcode := build_microcode_initrd(context):
+        initrds = [microcode]
+    else:
+        initrds = []
+
+    if context.config.initrds:
+        initrds += context.config.initrds
+    elif any((context.artifacts / "io.mkosi.initrd").glob("*.img")):
```

Comment:
We should drop the `img`. Everything in `io.mkosi.initrd` should be treated as an initrd I think.

Reply:
Or at least allow `.cpio` as well, and allow compression suffixes.

Reply:
I'll just drop it, makes more sense in light of already treating everything in the microcode directory as microcode